src/Sylius/Bundle/PayumBundle/DependencyInjection/SyliusPayumExtension.php

```diff
@@ -58,8 +58,10 @@ public function prepend(ContainerBuilder $container)
 
         $config = $container->getExtensionConfig('payum');
         $gateways = [];
-        foreach (array_keys($config[0]['gateways']) as $gatewayKey) {
-            $gateways[$gatewayKey] = 'sylius.payum_gateway.'.$gatewayKey;
+
+        foreach ($config[0]['gateways'] as $gatewayKey=>$config) {
```

Comment:
This takes only the first found configuration, while Configuration class allows us to define a way to merge a few of them into one.